app/src/main/java/org/coepi/android/ui/onboarding/OnboardingViewModel.kt

```diff
@@ -14,10 +27,33 @@ class OnboardingViewModel(
     private val preferences: Preferences
 ) : ViewModel() {
 
+    val context: Context = TODO("Need this to pick up main context, not sure how in VM");
+    val activity: Activity = TODO("Need this to attach to main activity, not sure how in VM");
     val text: LiveData<String> = Observable.just("TODO onboarding").toLiveData()
 
-    fun onCloseClick() {
-        preferences.putBoolean(SEEN_ONBOARDING, true)
+
+    fun onCtaClick() {
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.BLUETOOTH) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
Move this to an own component and wire with the other checks in `BlePreconditions`